Project Management Process.md

```diff
@@ -0,0 +1,287 @@
+    Milestone A     |     Milestone 2           Goal C
+    Milestone B     |       Goal 4              Goal D
+    Milestone C     |     Milestone 3      
+                    |       Goal 5      
+  Projects          |  
+    Project 1       |   Backlog             Backlog
+    Project 2       |     Goal 1              Goal A
+    Project 3       |     Goal 2              Goal B
+                    |     Goal 3              Goal C
+                    |     Goal 4              Goal D
+                    |     Goal 5
+```
+
+### Goals
+At the heart of each project are the goals. They are the basic unit of work. Goals can be anything that ***adds value to a release***. This includes issues reported by users, feature requests, bugs, Pull-Requests, general tasks such as refactoring, documentation or research. Goals are collected, maintained and tracked in the backlog by the project lead.
+
+A goal description includes the following:
+- Name / Summary
+  - Describes what the goal is on high level
+  - Example 1: "Reduce the size of the go-ipfs executable"
+  - Example 2: "js-ipfs should have API Documentation"
+- Description
+  - What the problem is and how to achieve this goal
+  - Should tell you how you will know that the problem has been solved or the goal has been achieved
+  - Describe the type of this goal: is it a bug? a new feature? something else?
+- Status
+  - The status should signal the goal's status towards completion
+  - What is the current state of this goal? Is it to-be-reviewed? In progress? Blocked? Done?
+  - A goal can be in only one state at a time
+- Dependencies (where applicable)
+  - Links to other goals that must happen before or concurrently with this one
+- Any additional information that helps to reach this goal (optional)
+
+TODO: a clear example
+
+### Backlog
+Backlog is the collection of all goals in a project. It should be as comprehensive as possible, including everything that needs to happen or should happen in a particular project. The backlog indicates priority by order: the items that have the highest priority are at the top of the list. The backlog is owned and maintained by the project lead. As part of daily work, the backlog gets updated and a general grooming should happen on weekly basis.
+
+A backlog has the following information:
+- All goals of a project
+- Priority of goals in sorted order
+
+While the backlog is usually one big list of goals, it may sometimes become convoluted. In this case, the backlog list can be split in two: ***Backlog*** and ***Future Work***. The future work section lists goals that will probably not be worked on anytime soon, or are not ready to be worked on yet, whereas the ready section contains the goals that are ready to be worked on or "approved". If split, together they are called "the backlog" while *Future Work* is referenced only in its specific meaning. The split and maintenance of both lists are done by the project lead.
```

Comment:
(oh wow, I totally remember writing this before but now i can't find it in prior comments so maybe i didn't...)

- i think it's confusing to call both "(a) all that needs to be done" and "(b) the things slated to be done soon" (i.e. `(a) - "Future Work"`  "Backlog". because we end up with `Backlog = Backlog - Future Work` which is _understandable_ (given how Future Work arose), but is confusing to newcomers.
- How about a new name `(a) Backlog` and `(b) Ready`, so that `Backlog = Future Work + Ready`. 
- We could use a different word than `Ready`.
- I have seen `Ready` used in other kanban-ish pipelines. I think "Ready" is nice because it makes it clear "this stuff is ready to be worked on". Other words like that may be: "Next", "Ready for Work", "Staged" (staged to go), "Primed", "Prepared", "All Systems Go" ([which is apparently a synonym of "Ready"](https://ipfs.io/ipfs/QmRW43LmQhpbUdH34PVJvuQ3ZjaN5eEqiDdGJgxQM5PzuQ/cap.png)) maybe there's others.